lib/astarte_device.ex

```diff
@@ -363,25 +366,28 @@ defmodule Astarte.Device do
     handle_disconnected_publish(from)
   end
 
-  def no_certificate(:internal, :request_certificate, data) do
+  def no_certificate(:internal, {:request_certificate, retry}, data) do
     case Impl.request_certificate(data) do
       {:ok, new_data} ->
-        actions = [{:next_event, :internal, :request_info}]
+        actions = [{:next_event, :internal, {:request_info, 0}}]
         {:next_state, :waiting_for_info, new_data, actions}
 
       {:error, :temporary} ->
-        # TODO: exponential backoff
-        actions = [{:state_timeout, 30_000, :retry_request_certificate}]
-        _ = Logger.warn("Trying again in 30 seconds")
+        backoff_time = :rand.uniform((1 <<< retry) * 1000)
```

Comment:
Randomizing the timeout to avoid the thundering herd problem is fine, but this specific implementation is "too random" (for example it doesn't guarantee that the backoff is growing monotonically).

A better approach is guaranteeing a minimum delay and then summing a randomized value, usually expressed as a percentage of the original delay. See for example how we handle this in the [Qt5 SDK](https://github.com/astarte-platform/astarte-device-sdk-qt5/blob/b51bcd8553978ca7aa986e66a3b8bb8cffabd073/hyperdrive/hyperdriveutils.cpp#L32)